### PR TITLE
Perform a robot unload on baton release

### PR DESCRIPTION
**Pull request by rtuck99**
Fixes 

* #1191 

Link to dodal PR (if required): #N/A
(remember to update `pyproject.toml` with the dodal commit tag if you need it for tests to pass!)

### Instructions to reviewer on how to test:

1. Do thing x
2. Confirm thing y happens

### Checks for reviewer

- [ ] Would the PR title make sense to a user on a set of release notes

### CODE REVIEW

File: tests/unit_tests/hyperion/test_baton_handler.py [hunk 1]
```diff
@@ -649,3 +683,92 @@ async def error_with_command_then_resume():
...
+    mock_load_centre_collect.side_effect = RuntimeError("Simulated beamline error")
+
+    with pytest.raises(PlanException):
+        run_udc_when_requested(bluesky_context, udc_runner)
+
+    mock_robot_unload.assert_not_called()
+
+
+@patch("mx_bluesky.hyperion.baton_handler.robot_unload")
+def test_robot_unload_still_performed_when_sample_exception(
```
Comment (DominicOram) [inline tests/unit_tests/hyperion/test_baton_handler.py:755]: Good catch of edge case, thanks!

**Review by DominicOram — APPROVED** [review]
Great, thank you!
